src/local-history/local-history-manager.ts

```diff
@@ -27,6 +27,11 @@ export class LocalHistoryManager {
         vscode.workspace.onDidSaveTextDocument((document: vscode.TextDocument) => {
             this.saveEditorContext(document);
         });
+        vscode.workspace.onDidChangeTextDocument((document: vscode.TextDocumentChangeEvent) => {
```

Comment:
I think this `event` would be too costly (it is triggered whenever a file has been modified (each character)).
Instead, I think we should simplify the behavior to control the amount of times we write local-history to disk when listening to `onDidSaveTextDocument`.